enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,366 @@
+deployments, it will provide the same integration with OpenStack services (e.g.
+storage, load balancer).
+
+The focus will be on adding baremetal nodes to an existing (virtualised)
+OpenShift cluster as a day two operation. Deploying baremetal workers during
+the initial installation can be implemented later.
+
+## Motivation
+
+In heavy-loaded clusters, or when the workload is better handled with specially
+purposed hardware, the functional requirement of controlling where the work is
+computed might outweight the added value of virtualisation.
+
+In OpenStack clusters where Nova transparently provisions baremetal machines,
+this enhancement will ensure that OpenShift installations can use them as
+Compute Nodes.
+
+Existing OpenShift clusters running on a baremetal-enabled OpenStack will be
+able to add baremetal workers by adding a new node pool as a day-two operation.
+
+### Goals
+
+* Enable running OpenShift compute nodes on a mix of virtual and baremetal
+  machines on an OpenStack tenant.
+* Provide the same integration with the underlying OpenStack services as in the fully virtualised environment.
+* Baremetal machines can be added and removed from a running OpenShift cluster using the Machine API.
+* Support both `flat` and `neutron` Ironic Network Interfaces
+  - https://docs.openstack.org/ironic/queens/admin/multitenancy.html
+  - A worker node pool will need to be able to specify a Neutron network in
+    addition to a Nova flavor
+
+### Non-Goals
+
+**Baremetal bootstrap and control plane nodes.**
+
+This proposal focuses on the OpenShift compute nodes only. Deploying or
+migrating the control plane to bare metal is out of scope.
+
+It may be added later as a separate feature.
+
+**Baremetal deployments on anything other than an OpenStack tenant**
+
+There are other efforts that provide OpenShift on the bare metal. See
+[Metal³][metal3-io] and the [Baremetal installer-provisioned (IPI) /
+user-provisioned (UPI) infrastructure support in the OpenShift Installer].
+
+These projects allow deploying OpenShift in an environment where all the nodes
+are physical and there is no OpenStack or other cloud layer in-between.
+
+In contrast, this proposal uses the existing OpenStack provider and extends it
+to allow adding baremetal workers if the cloud supports it.
+
+**Attaching floating IPs to the BM workers.**
+
+The OpenStack cluster might make the baremetal nodes accessible to the outside
+world, but exposing them via floating IPs might represent an increase in
+complexity and is currently not an explicit goal.
+
+**Provisioning baremetal workers at install time.**
+
+To simplify the necessary development work, testing and reduce the time
+required to provide a useful feature, we will not focus deploying baremetal
+workers during the initial OpenShift install phase.
+
+That would require changes to the installer and adding options to the
+`install-config.yaml` format.
```

Comment:
This sentence and the one prior are a bit confusing.  They need some rewording. something like: 
"Provisioning baremetal workers at install time would  require changes to the installer and other work which would increase the complexity of the development work.  A useful feature can be delivered without this functionality.

Reply:
Removed, as we now want to target provisioning at install time